tools/StandardAnchorTags/Program.cs

```diff
@@ -11,10 +14,20 @@ public class Program
     private const string FilesPath = "../standard/clauses.json";
     private const string GrammarFile = "grammar.md";
 
-    private static Clauses? standardClauses;
-
-    static async Task<int> Main(string[] args)
+    private static Clauses? standardClauses = null!;
+
+    /// <summary>
+    /// Update section numbers, anchor tags, and the TOC
+    /// </summary>
+    /// <param name="owner">The GitHub owner org (for example, "dotnet")</param>
+    /// <param name="repo">The GitHub repo name (for example, "csharpstandard")</param>
+    /// <param name="headSha">The commit sha, when run as a GitHub action</param>
+    /// <param name="dryrun">True for a dry run, false to update the text in all files</param>
+    /// <returns>0 on success, non-zero on failure</returns>
+    static async Task<int> Main(string owner, string repo, string? headSha = null, bool dryrun =false)
```

Comment:
There's a lot less actual changes than appears in this file. The `logger` is new. And, it's passed to the objects that do the work. Finally, it logs the errors and uploads the collection of annotations.

The extra diffs are because the `Main` method was getting long, so I broke it into 3 distinct functions.